backend/app/controllers/api/v1/roles_controller.rb

```diff
@@ -2,6 +2,6 @@ class Api::V1::RolesController < Api::V1::ApiController
   skip_before_action :authenticate_user!, only: [:index]
 
   def index
-    render json: Role.all
+    render json: current_user&.admin? ? Role.all : Role.where.not(role_type: Role::ADMIN)
```

Comment:
create scope for `Role` class like `by_user`

Reply:
Oh, yea makes sense!